### PR TITLE
absorb batchnorm

### PR DESCRIPTION
**Pull request by eladhoffer**

### CODE REVIEW

File: torch/nn/utils/__init__.py [hunk 1]
```diff
@@ -1,2 +1,3 @@
 from . import rnn
 from .clip_grad import clip_grad_norm
+from .absorb_bn import absorb_bn, search_absorbe_bn
```
Comment (ngimel) [inline torch/nn/utils/__init__.py]: Is absorb**e** a typo, or does it have a special meaning?

**Comment by ngimel** [comment]
Cool! And special batchnorm backprop in inference mode would no longer be necessary, as there in no batchnorm in inference mode, just scale and shift.

File: torch/nn/utils/absorb_bn.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    bn_module.register_buffer('running_mean', None)
+    bn_module.register_buffer('running_var', None)
+    bn_module.register_parameter('weight', None)
+    bn_module.register_parameter('bias', None)
+    bn_module.affine = False
+
+
+def is_bn(m):
+    return isinstance(m, nn.BatchNorm2d) or isinstance(m, nn.BatchNorm1d)
```
Comment (ngimel) [inline torch/nn/utils/absorb_bn.py]: Then Conv3d should be added to is_absorbing?

**Comment by soumith** [comment]
this PR needs rework and is very limited to certain model cases. I'm closing it for the lack of activity anyways.
